### PR TITLE
Enable instantiation evaluation checks by default

### PR DESCRIPTION
**Pull request by ajreynol**
Makes `--ieval=use` the default, as it improves both SMT-LIB and TPTP quantifiers.

This PR makes a few minor changes to `ieval` to make it match the previous behavior for (non-incremental) entailment checks. In particular a corner case of UF and nested quantified formulas were leading to spurious failures.

### CODE REVIEW

**Comment by ajreynol** [comment]
+156-52 on unsat TPTP:
```
                           Configuration  #unsat      time    #sat      time #solved    #total
                      cvc5-050923-ievt_d    9809   10253.6     162   6.78192      9971     22011
                  cvc5-050923-ievt_ieval    9913   10838.3     163   6.87383     10076     22011

For unsatisfiable : 
                      cvc5-050923-ievt_d       X      52
                  cvc5-050923-ievt_ieval     156       X
```
Pending SMT-LIB quantifiers results.

**Comment by ajreynol** [comment]
SMT-LIB is +187-99:
```
                           Configuration  #unsat      time    #sat      time #solved    #total
                      cvc5-050923-ievt_d   63787   20792.3    1533   375.618     65320     91390
                  cvc5-050923-ievt_ieval   63875   21775.9    1533   376.178     65408     91390



For unsatisfiable : 
                      cvc5-050923-ievt_d       X      99
                  cvc5-050923-ievt_ieval     187       X

File: src/theory/quantifiers/ieval/term_evaluator.cpp [hunk 1]
```diff
@@ -45,6 +45,10 @@ TermEvaluatorEntailed::TermEvaluatorEntailed(Env& env,
 
 TNode TermEvaluatorEntailed::evaluateBase(const State& s, TNode n)
 {
+  if (n.getKind()==FORALL)
```
Comment (HanielB) [inline src/theory/quantifiers/ieval/term_evaluator.cpp:48]: format

**Review by HanielB — APPROVED** [review]
Nice!

**Pull request merged**